go-controller/pkg/ovn/policy.go

```diff
@@ -959,6 +959,14 @@ func (oc *Controller) addNetworkPolicy(policy *knet.NetworkPolicy) {
 		return
 	}
 
+	type policyHandler struct {
+		gress             *gressPolicy
+		namespaceSelector *metav1.LabelSelector
+		podSelector       *metav1.LabelSelector
+		addrSet           string
+		peerMap           map[string]bool
+	}
```

Comment:
this could maybe just all be added to `gressPolicy`?

Reply:
@danwinship yeah there are a bunch of improvements (like the one you mentioned about watchers yesterday) to this code and it is going to be refactored anyway. I don't want to make changes other than what is needed to just fix this locking bug in this commit.